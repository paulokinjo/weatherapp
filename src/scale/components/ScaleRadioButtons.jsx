import { CELSIUS, FAHRENHEIT } from '../utils/constants';

import FormControlLabel from '@material-ui/core/FormControlLabel';
import Grid from '@material-ui/core/Grid';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import React from 'react';

const ScaleRadioButtons = ({ onChange }) => {
  return (
    <RadioGroup
      row
      aria-label="scales"
      name="scales"
      defaultValue={FAHRENHEIT}
      onChange={onChange}
    >
      <Grid item container xs={12}>
        <Grid item container xs={false} sm={2} />
        <Grid item container xs={8} sm={6}>
          <FormControlLabel
            value={CELSIUS}
            control={<Radio color="primary" />}
            label="Celsius"
            labelPlacement="end"
          />
        </Grid>
        <Grid item container xs={4} sm={2}>
          <FormControlLabel
            value={FAHRENHEIT}
            control={<Radio color="primary" />}
            label="Fahrenheit"
            labelPlacement="end"
          />
        </Grid>
        <Grid item container xs={false} sm={2} />
      </Grid>
    </RadioGroup>
  );
};

export default ScaleRadioButtons;
