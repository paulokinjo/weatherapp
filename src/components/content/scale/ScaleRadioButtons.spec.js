import React from 'react';
import ScaleRadioButtons from './ScaleRadioButtons';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

describe('ScaleRadioButtons', () => {
  const scaleRadioButtons = shallow(<ScaleRadioButtons />);

  describe('Layout', () => {
    it('should render properly', () => {
      expect(scaleRadioButtons).toMatchSnapshot();
    });
  });

  describe('Lifecycle', () => {
    it('should initialize Fahrenheit as default value', () => {
      const { queryByLabelText } = render(<ScaleRadioButtons />);
      const fahrenheitRadioButton = queryByLabelText('Fahrenheit');

      expect(fahrenheitRadioButton.checked).toBe(true);
    });

    it('should have Celsius as unchecked', () => {
      const { queryByLabelText } = render(<ScaleRadioButtons />);
      const celsiusRadioButton = queryByLabelText('Celsius');

      expect(celsiusRadioButton.checked).toBe(false);
    });
  });
});
