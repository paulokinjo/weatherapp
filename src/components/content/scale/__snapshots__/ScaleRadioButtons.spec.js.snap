// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScaleRadioButtons Layout should render properly 1`] = `ShallowWrapper {}`;
